packages/fetchai/connections/p2p_libp2p/libp2p_node/acn/utils.go

```diff
@@ -48,25 +78,25 @@ func DecodeAcnMessage(buf []byte) (string, *AeaEnvelope, *Status, error) {
 		return msg_type, nil, nil, err
 	}
 	// response is either a LookupResponse or Status
-	var aeaEnvelope *AeaEnvelope = nil
-	var status *Status = nil
+	var aeaEnvelope *AeaEnvelopePerformative = nil
+	var status *StatusBody = nil
 
-	switch pl := response.Payload.(type) {
-	case *AcnMessage_AeaEnvelope:
+	switch pl := response.Performative.(type) {
+	case *AeaEnvelope:
 		aeaEnvelope = pl.AeaEnvelope
 		msg_type = "aea_envelope"
-	case *AcnMessage_Status:
-		status = pl.Status
+	case *Status:
+		status = pl.Status.Body
 		msg_type = "status"
 	default:
-		err = errors.New("unexpected ACN Message")
+		err = fmt.Errorf("unexpected ACN Message: %s", response)
 		logger.Error().Msg(err.Error())
 		return msg_type, nil, nil, err
 	}
```

Comment:
should this case be handled as `UNEXPECTED_PAYLOAD`? Currently, it is handled as `ERROR_SERIALIZATION`

Reply:
`UNEXPECTED_PAYLOAD` should be handled outside this function. In this function, we are not only decoding as the function signature suggests, but also deciding whether we can handle it.

Reply:
updated